gravitee-am-test/specs/gateway/account-link.jest.spec.ts

```diff
@@ -0,0 +1,330 @@
+    name: 'app-local',
+    type: 'WEB',
+    clientId: 'app-local',
+    clientSecret: 'app-local',
+    redirectUris: ['https://test.com'],
+  }).then((app) =>
+    updateApplication(
+      domainAccLinking.id,
+      accessToken,
+      {
+        identityProviders: [{ identity: localIdp.values().next().value.id, priority: -1 }],
+      },
+      app.id,
+    ),
+  );
+  domainUser = getUser('domain-user');
+  // Create the same user
+  await createUser(domainAccLinking.id, accessToken, domainUser);
+  await delay(6000);
+});
+
+describe('Account Linking - local IDP and OIDC', () => {
+  it('Should login and create double user - account linking not setup', async () => {
+    const user1TokenResponse = await loginUserNameAndPassword(
+      applicationAccLinking.settings.oauth.clientId,
+      oidcUser,
+      'Test1234567!',
+      false,
+      domainAccLinkingConfiguration,
+      domainAccLinking,
+      'https://test.com',
+      'code',
+      true,
+    );
+    expect(user1TokenResponse.headers['location']).toContain('test.com?code=');
+    const allUsers = await getAllUsers(domainAccLinking.id, accessToken);
+    expect(allUsers.data.length).toBe(2);
+    // Remove and recreate users
+    await deleteUser(domainAccLinking.id, accessToken, allUsers.data[0].id);
+    await deleteUser(domainAccLinking.id, accessToken, allUsers.data[1].id);
+    await createUser(domainAccLinking.id, accessToken, domainUser);
+  });
+
+  it('Should login and create double user - account linking setup', async () => {
+    const flows = await getDomainFlows(domainAccLinking.id, accessToken);
+    lookupFlowAndResetPolicies(flows, 'CONNECT', 'pre', [
+      {
+        name: 'Account Linking',
+        policy: 'policy-am-account-linking',
+        description: '',
+        configuration: JSON.stringify({
+          exitIfNoAccount: false,
+          exitIfMultipleAccount: false,
+          userAttributes: [{ name: 'email', value: 'test@test.fr' }],
+        }),
+        enabled: true,
+        condition: '',
+      },
+    ]);
+    await updateDomainFlows(domainAccLinking.id, accessToken, flows);
+    await delay(6000);
+    const user1TokenResponse = await loginUserNameAndPassword(
+      applicationAccLinking.settings.oauth.clientId,
+      oidcUser,
+      'Test1234567!',
+      false,
+      domainAccLinkingConfiguration,
+      domainAccLinking,
+      'https://test.com',
+      'code',
+      true,
+    );
+    expect(user1TokenResponse.headers['location']).toContain('test.com?code=');
+    cookies = user1TokenResponse.headers['set-cookie'];
+    const allUsers = await getAllUsers(domainAccLinking.id, accessToken);
+    expect(allUsers.data.length).toBe(1);
+  });
+
+  it('Should login to second app with local idp', async () => {
+    const authResponse = await performGet(
+      domainAccLinkingConfiguration.authorization_endpoint,
+      '?response_type=code&client_id=app-local&redirect_uri=https://test.com',
```

Comment:
suggestion: we can use the object here to provide the clientId instead of hardcoded value so it will be more simple the change value if we need to.